lib/rspec/buildkite/analytics/session.rb

```diff
@@ -177,11 +172,53 @@ def connect
       wait_for_confirm
 
       @logger.write("connected")
+
+      # As this connect method can be called multiple times in the
+      # reconnection process, kill prev write threads (if any) before
+      # setting up the new one
+      @write_thread&.kill
+
+      @write_thread = Thread.new do
+        @logger.write("hello from write thread")
+        # Pretty sure this while loop is fine cos the call to queue.pop is blocking
+        while
```

Comment:
Instead of a `while` without condition we can use `loop`
```suggestion
        loop
```

Reply:
should that be `loop do`?